### ISSUE TITLE
ARM Support?

### ISSUE THREAD

**Issue by drautb** [papertrail/remote_syslog2#109]
I'd like to use this on a Raspberry Pi. I'm willing to attempt building it myself, but it would super nice if there were a binary for ARM architectures too.

**Comment by rephus** [comment]
+1
I'd prefer not to install and setup golang (or any other dependency) on my Pi just to build this.

**Comment by troy** [comment]
Anyone have experience cross-compiling to ARM (from Linux) for the Pi and/or Pi 2? Some docs imply that just setting `GOARM=6` is enough, [other docs](https://5pi.de/2015/02/10/prometheus-on-raspberry-pi/) say, well, otherwise :-)

**Comment by leonsodhi** [comment]
It might be as easy as running `./all.bash` (any other steps needed are listed [here](http://dave.cheney.net/2012/09/08/an-introduction-to-cross-compilation-with-go)), changing the arch to `arm`, and then removing the `-cgo` switch. I believe it's even easier on Go 1.5, but r_s2 won't currently build on it so we're using 1.3 for the moment.

I don't have an ARM system to test on but did manage to produce a standard [ARM binary](https://www.dropbox.com/s/yfjoi847nkikl6a/remote_syslog_linux_arm_default_016.tar.gz?dl=0) and a [GOARM=5 binary](https://www.dropbox.com/s/oqapflt6esqyw72/remote_syslog_linux_arm_goarm5_016.tar.gz?dl=0).

**Comment by jotadepicas** [comment]
@leonsodhi I've just set it up on a raspberry pi B 3.18.9+ armv6l Raspbian wheezy and I can confirm it works ok. Thanks!

**Comment by leonsodhi** [comment]
Thanks for confirming, @jotadepicas. Did you use the `GOARM=5` binary, the other one, or maybe you built something yourself?

**Comment by johlym** [comment]
I'm going to think about testing this in 0.18b2, maybe b3. I don't really have any devices to test this on so I don't know if it'll make it to the GM.

**Comment by johlym** [comment]
Will add it to the 0.18b2 milestone for now.

**Comment by johlym** [comment]
Added ARM builds to 0.18b1 release. These are 100% UNTESTED at this time. YMMV.

**Comment by MonicaG** [comment]
Hi,

I am getting the following error when I install the "remote-syslog2_0.18_arm.deb" package:

command: sudo dpkg -i remote-syslog2_0.18_arm.deb

error:
dpkg: error processing archive remote-syslog2_0.18_arm.deb (--install):
 package architecture (arm) does not match system (armhf)
Errors were encountered while processing:
 remote-syslog2_0.18_arm.deb

I am using a raspberry pi 3.

**Comment by MonicaG** [comment]
Hi,

I tried the [remote_syslog_linux_arm.tar.gz](https://github.com/papertrail/remote_syslog2/releases/download/v0.18-beta1/remote_syslog_linux_arm.tar.gz) and it works!  I was able to set it up and see log messages appearing on my dashboard.

Thank you @johlym! 😄

**Comment by johlym** [comment]
Hi @MonicaG!

That's great news! The `.deb` and `.rpm` packages themselves are still pretty flaky so I'm honestly not surprised that it didn't work. I'm super happy to see the ARM binary is actually functional, though :) 

I'm going to leave this open until I can test and verify on my own, especially those packages. 

Thanks, again! Have a great weekend.

**Comment by johlym** [comment]
@MonicaG If you're up for it, would you be interested in attempting to add `arm` architecture support?

```
dpkg --add-architecture arm
```

And retry the `dpkg` install?

**Comment by MonicaG** [comment]
Hi @johlym 

I followed the instructions you gave above on my raspberry pi and the install worked!  I did a quick test and verified that log messages were showing up on the papertrail dashboard.

Thank you! 😄

**Comment by johlym** [comment]
@MonicaG Thanks for that :) I really appreciate it. I'll have an RPI next week so I would have discovered this eventually. 

Which OS on your RPI 3 are you using?

**Comment by MonicaG** [comment]
Hi @johlym,

I am using the [raspbian - jessie](https://www.raspberrypi.org/downloads/raspbian/) os, which I installed with NOOBS.

One side effect I noticed from using the ```dpkg --add-architecture arm``` command was I received errors when updating my pi.  When I did ```sudo apt-get update``` I got errors stating:
> 'Unable to find expected entry 'main/binary-arm/Packages''  

I was able to get around this by specifying ```[ arch=armhf ]``` in my /etc/apt/sources.list.d/raspi.list and /etc/apt/sources.list files.  Example:
```
deb [arch=armhf] http://mirrordirector.raspbian.org/raspbian/ jessie main contrib non-free rpi
```
Is it possible to compile for the armhf architecture to avoid this?

**Comment by johlym** [comment]
@MonicaG thanks for making me aware of the hidden danger of `--add-architecture arm`. :)

I _think_ it's possible but honestly I can't say for sure, yet, as I'm writing this before I've tried. I'll be able to find out in the next week or so. Stay tuned.

**Comment by johlym** [comment]
I'm able to build with both ARM and ARMHF as platforms. I'm not entirely sure if there's a difference in the build process or the codebase. I know non-HF platforms lack a hardware floating point instructions (hence the lack of HF) but what that means for the binary, I can only guess. Modern non-HF boards that are reasonable to use seem to be harder to get my hands on.

**Comment by johlym** [comment]
Fixed in #160

**Comment by MonicaG** [comment]
@johlym - Cool! Thank you 😄
